mesonbuild/compilers/cuda.py

```diff
@@ -309,13 +309,16 @@ def get_xcompiler_val(flag: str, flagit: T.Iterator[str]) -> str:
                 raise ValueError("-Xcompiler flag merging failed, unknown argument form!")
         return xflags
 
-    def _to_host_flags(self, flags: T.List[str], phase: _Phase = _Phase.COMPILER) -> T.List[str]:
+    @classmethod
+    def _to_host_flags_base(cls, flags: str | T.List[str], phase: _Phase = _Phase.COMPILER, default_include_dirs: T.Optional[T.List[str]] = None) -> T.List[str]:
```

Comment:
If you're going to have `list[str] | str` you might as well just do `T.Sequence[str]`, which would cover both.

Reply:
Which we shouldn't, really. Instead in detect.py I'd expect to see it split eagerly and passed as a list into this function.